### PR TITLE
Update to 3.0 preview5

### PR DESCRIPTION
**Pull request by crummel**

### CODE REVIEW

**Comment by crummel** [comment]
Might still be some minor stuff to clean up in CI but I think this is ready for review.

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -1,59 +1,59 @@
...
+      <Sha>79b8271f65609b23e0c0ee1babed6862b7c5ed56</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Build" Version="16.1.0-preview.27">
+    <Dependency Name="Microsoft.Build" Version="16.0.462">
       <Uri>https://github.com/microsoft/msbuild</Uri>
-      <Sha>d004974104fde202e633b3c97e0ece3287aa62f9</Sha>
+      <Sha>62fb89029dac66f73a5ca52363fd11b36f12190b</Sha>
     </Dependency>
-    <Dependency Name="NuGet.Build.Tasks" Version="5.0.0-preview1.5754">
+    <Dependency Name="NuGet.Build.Tasks" Version="5.1.0-rtm.5921">
```
Comment (dseefeld) [inline eng/Version.Details.xml]: How is it that this version number changes but the Sha remains the same?

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -1,59 +1,59 @@
...
       <RepoName>nuget-client</RepoName>
     </Dependency>
-    <Dependency Name="Microsoft.NET.Sdk" Version="3.0.100-preview4.19165.1">
+    <Dependency Name="Microsoft.NET.Sdk" Version="3.0.100-preview5.19225.2">
       <Uri>https://github.com/dotnet/sdk</Uri>
-      <Sha>814b7898f9908a88f62706331cf56f1ecc9745eb</Sha>
+      <Sha>7af4517bbcf17b000889c1077ef5976f2b6350ea</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.DotNet.Cli.CommandLine" Version="1.0.0-preview.19127.1">
+    <Dependency Name="Microsoft.DotNet.Cli.CommandLine" Version="1.0.0-preview.19208.1">
```
Comment (dseefeld) [inline eng/Version.Details.xml]: Same here.  The version changes but the Sha doesn't.  What is this Version used for?

File: eng/Version.Details.xml [hunk 3]
```diff
@@ -105,7 +105,7 @@
       <Uri>https://github.com/dotnet/cli-migrate</Uri>
       <Sha>37e3a01aa1df320a286b0a858876f9994fb6bb18</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Extensions.CommandLineUtils" Version="1.0.1">
+    <Dependency Name="Microsoft.Extensions.CommandLineUtils" Version="3.0.0-preview5.19227.9">
```
Comment (dseefeld) [inline eng/Version.Details.xml]: Another case of Version changes, Sha doesn't.

File: patches/sdk/0002-Add-missing-PlatformAbstractions-reference.patch [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
+index 41bbd113..848504be 100644
 --- a/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj
 +++ b/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj
-@@ -44,6 +44,7 @@
+@@ -45,6 +45,7 @@
      <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkVersion)" ExcludeAssets="Runtime" />
      <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCoreVersion)" ExcludeAssets="Runtime" />
      <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="$(MicrosoftExtensionsDependencyModelVersion)" />
 +    <PackageReference Include="Microsoft.DotNet.PlatformAbstractions" Version="$(MicrosoftDotNetPlatformAbstractionsPackageVersion)" />
+     <PackageReference Include="Microsoft.NET.HostModel" Version="$(MicrosoftNETHostModelVersion)" />
```
Comment (dseefeld) [inline patches/sdk/0002-Add-missing-PlatformAbstractions-reference.patch]: Will this add a new prebuilt?

File: repos/core-sdk.proj [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
   <PropertyGroup>
-    <PreferredHash>118dd862c853ff39694449ac303f5d8ef7d11b24</PreferredHash>
+    <PreferredHash>b487ff10aafb8b58eaa2b78a236b1e97999d7a22</PreferredHash>
   </PropertyGroup>
 
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))/dir.props" />
 
   <PropertyGroup>
-    <OfficialBuildId>20190416.21</OfficialBuildId>
+    <OfficialBuildId>fix me</OfficialBuildId>
```
Comment (dseefeld) [inline repos/core-sdk.proj]: This seems like it needs an update.

File: eng/Version.Details.xml (hunk 1 above)
Comment (crummel) [reply eng/Version.Details.xml]: They were not participating in the prodcon build, rebased on your NuGet update.

File: eng/Version.Details.xml (hunk 2 above)
Comment (crummel) [reply eng/Version.Details.xml]: This has been rebuilt a few times but has no code changes, so the SHA never changed.

File: eng/Version.Details.xml (hunk 3 above)
Comment (crummel) [reply eng/Version.Details.xml]: This was a result of aspnet/common moving to Arcade, which changes their version number algorithm.  No code changes.

File: patches/sdk/0002-Add-missing-PlatformAbstractions-reference.patch (hunk 4 above)
Comment (crummel) [reply patches/sdk/0002-Add-missing-PlatformAbstractions-reference.patch]: Yes, but this is existing code being moved (see https://github.com/dotnet/core-setup/issues/6338).

File: repos/core-sdk.proj (hunk 5 above)
Comment (crummel) [reply repos/core-sdk.proj]: Thanks for the catch, added the actual build and version numbers.

**Pull request merged**